src/Domain/ElementDistribution.cpp

```diff
@@ -0,0 +1,154 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/ElementDistribution.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+#include "Domain/Structure/ElementId.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace domain {
+
+// This interleaves the bits of the element index.
+// A sketch of a 4x2 2D element, with bit indices and resulting z-curve
+//
```

Comment:
According to Wikipedia, this should be called a z-order curve. A z-curve is something in bioinformatics. Since I'd never heard of either of these things, feel free to disregard.